pipeline_steps/gate.groovy

```diff
@@ -0,0 +1,194 @@
+                job = Hudson.instance.getItems(org.jenkinsci.plugins.workflow.job.WorkflowJob).find {it.displayName == jobName}
+                nextNumber = job.getNextBuildNumber()
+                println("Requesting new build:")
+                build(
+                    job: jobName,
+                    wait: false,
+                    parameters: [
+                        [
+                            $class: "StringParameterValue",
+                            name: "RPC_GATING_BRANCH",
+                            value: RPC_GATING_BRANCH,
+                        ],
+                        [
+                            $class: "StringParameterValue",
+                            name: "BRANCH",
+                            value: ghprbTargetBranch,
+                        ],
+                        [
+                            $class: "StringParameterValue",
+                            name: "pullRequestChain",
+                            value: common.dumpCSV(pullRequestIDs),
+                        ],
+                    ]
+                )
+                while(true){
+                    sleep(time: 20, unit: "SECONDS")
+                    potentialBuild = job.getLastBuild()
+                    found = false
+                    while(potentialBuild.getNumber() >= nextNumber){
+                        potentialBuildCause = potentialBuild.getCauses()[0]
+                        if (potentialBuildCause instanceof Cause.UpstreamCause){
+                            if (potentialBuildCause.getUpstreamBuild() == triggerBuild.getNumber()){
+                                found = true
+                                gateBuilds.add(potentialBuild)
+                                break
+                            }
+                        }
+                        potentialBuild = potentialBuild.getPreviousBuild()
+                    }
+                    if (found){
+                        break
+                    }
+                }
+            }
+        }
+        println("Tests currently being tracked for gate:\n${gateBuilds.collect() {it.getAbsoluteUrl()}.join("\n")}")
+        if (
+                antecedentTriggerBuilds.empty
+                && gateBuilds.every {! it.hasntStartedYet() && ! it.isBuilding() && it.getResult() && it.getResult().equals(success) }
+                && gateJobNames == (gateBuilds.collect {it.getParent().getName()}).sort(false)
+        ) {
+            break
+        }else {
+            sleep(time: 120, unit: "SECONDS")
+        }
+    }
+
+    // Not sure where to get the context from so definition duplicated from job definition
+    statusContext = "CIT/gate"
+    (pr_repo_org, pr_repo_name) = ghprbGhRepository.split("/")
+    description = "Gate tests passed, merging..."
+    github.create_status(pr_repo_org, pr_repo_name, ghprbActualCommit, "success", triggerBuild.getAbsoluteUrl(), description, statusContext)
```

Comment:
A comment in here to say that this allows the PR to merge when `CIT/gate` is a required context in github configuration. Also that this does duplicate the status setting that GHPRB will do at the end of the job, but that github doesn't mind the same status/context getting set multiple times.